pkg/util/proxyutil/proxyutil.go

```diff
@@ -42,12 +44,28 @@ func PrepareProxyRequest(req *http.Request) {
 }
 
 // ClearCookieHeader clear cookie header, except for cookies specified to be kept (keepCookiesNames) if not in skipCookiesNames.
-func ClearCookieHeader(req *http.Request, keepCookiesNames []string, skipCookiesNames []string) {
+func ClearCookieHeader(req *http.Request, allowedCookies datasources.AllowedCookies, skipCookiesNames []string) {
 	keepCookies := map[string]*http.Cookie{}
-	for _, c := range req.Cookies() {
-		for _, v := range keepCookiesNames {
-			if c.Name == v {
-				keepCookies[c.Name] = c
+
+	if allowedCookies.MatchOption == datasources.MO_EXACT_MATCH {
+		for _, c := range req.Cookies() {
+			for _, v := range allowedCookies.KeepCookies {
+				if c.Name == v {
+					keepCookies[c.Name] = c
+				}
+			}
+		}
+	}
+
+	if allowedCookies.MatchOption == datasources.MO_REGEX_MATCH {
+		for _, c := range req.Cookies() {
+			re, err := regexp.Compile(allowedCookies.MatchPattern)
```

Comment:
Potential security issue allowing regexp from user input?

Reply:
I do not have enough knowledge about potential security issues. Could you elaborate on this, please? 

Allowing a star (`*`) would definitely solve what I want to achieve. But wouldn't that also bring security flaws?

Reply:
Maybe https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html#regular-expressions

> Allowing a star (*) would definitely solve what I want to achieve. But wouldn't that also bring security flaws?

We (grafana) can intercept each star (*) and replace that with the actual regexp/logic we want to apply and therefore it's not really regexp user input and safer.

Reply:
That's a good idea. I will implement it that way. thanks for the suggestion.